js/engine/src/index.d.ts

```diff
@@ -16,5 +16,8 @@ export interface Engine {
   ) => string;
   writeFile: (filename: string, content: string) => void;
   deleteFile: (filename: string) => void;
+  isMissingLanguages: () => boolean;
```

Comment:
Would be good to document a bit those methods. One of the thing I love the most about OCaml is the presence of those .mli signature files that let you document clearly the interface of a module. Then user of the module don't eve need to open the implementation file; just reading the signature and the comments in it (and types) should be clear enough.